packages/api/endpoints/rules.js

```diff
@@ -78,11 +78,11 @@ async function post(req, res) {
   const apiRule = req.body || {};
   const name = apiRule.name;
 
-  if (await ruleModel.exists(name)) {
-    return res.boom.conflict(`A record already exists for ${name}`);
-  }
-
   try {
+    if (await ruleModel.exists(name)) {
+      return res.boom.conflict(`A record already exists for ${name}`);
+    }
+
```

Comment:
I changed it because in the `catch` block we have:

```
    if (isBadRequestError(error)) {
      return res.boom.badRequest(error.message);
    }
```

And `isBadRequestError` checks to see if the error is of an expected class:

```
const isBadRequestError = (error) =>
  error.name === 'SchemaValidationError'
  || error.name === 'ValidationException'
  || error instanceof ValidationError
  || isPostgresValidationError(error)
  || isDynamoValidationException(error);
```

So I want to make sure the error is caught and will pass through the `isDynamoValidationException` helper that I added, so that the expected status code is returned.

Reply:
I'm missing something here.  the code you moved into the catch block either returns from the function or does nothing right?

Reply:
I was not considering errors thrown by `ruleModel.exists(name)` and while the tests appear to still pass, this is better.